pkg/envoy/ciliumenvoyconfig.go

```diff
@@ -157,18 +159,10 @@ func ParseResources(cecNamespace string, cecName string, anySlice []cilium_v2.XD
 				listener.EnableReusePort = &wrapperspb.BoolValue{Value: false}
 			}
 
-			// Inject Cilium bpf metadata listener filter, if not already present.
-			found := false
-			for _, lf := range listener.ListenerFilters {
-				if lf.Name == "cilium.bpf_metadata" {
-					found = true
-				}
-			}
-			if !found {
-				listener.ListenerFilters = append(listener.ListenerFilters, getListenerFilter(false /* egress */, useOriginalSourceAddr, isL7LB))
-			}
-			// Inject listener socket option for Cilium datapath
-			listener.SocketOptions = append(listener.SocketOptions, getListenerSocketMarkOption(false /* egress */))
+			// Figure out if this is an internal listener
+			internalListener := listener.GetAddress() != nil && listener.GetAddress().GetEnvoyInternalAddress() != nil
+			// Only inject Cilium filters if Cilium allocates listener address
+			injectCiliumFilters := listener.GetAddress() == nil && !internalListener
```

Comment:
Isn't the extra boolean check moot? We already know `internalListener` must be false because `listener.GetAddress()` is false.
```suggestion
			injectCiliumFilters := listener.GetAddress() == nil
```